src/config.rs

```diff
@@ -290,6 +296,54 @@ impl Config {
     }
 }
 
+#[derive(Debug, Clone, Copy, PartialEq, Serialize, Deserialize)]
+pub enum ClippyPreference {
+    /// Disable clippy
+    Off,
+    /// Enable clippy, but "allow" clippy lints (ie require "warn" override)
+    OptIn,
+    /// Enable clippy
+    On,
+}
+
+/// Permissive deserialization for `ClippyPreference`
+/// "opt-in", "Optin" -> `ClippyPreference::OptIn`
+impl FromStr for ClippyPreference {
+    type Err = ();
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().as_str() {
+            "off" => Ok(ClippyPreference::Off),
+            "optin" | "opt-in" => Ok(ClippyPreference::OptIn),
+            "on" => Ok(ClippyPreference::On),
+            _ => Err(()),
+        }
+    }
+}
+
+/// Permissive custom deserialization for `ClippyPreference` using `FromStr`
+fn deserialize_clippy_preference<'de, T, D>(deserializer: D) -> Result<T, D::Error>
+where
+    T: Deserialize<'de> + FromStr<Err = ()>,
+    D: Deserializer<'de>,
+{
+    struct ClippyPrefDeserializer<T>(PhantomData<fn() -> T>);
+
+    impl<'de, T> Visitor<'de> for ClippyPrefDeserializer<T>
+    where
+        T: Deserialize<'de> + FromStr<Err = ()>,
+    {
+        type Value = T;
+        fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+            formatter.write_str(r#"on", "opt-in" or "off""#)
```

Comment:
Is there a missing `"` before `on` -> `r#""on"`

Reply:
Yeah, nice catch.